apis/v1alpha1/cluster_tc_program_types.go

```diff
@@ -37,25 +37,27 @@ type ClTcAttachInfo struct {
 	// program. If Containers is not specified, the BPF program will be attached
 	// in the root network namespace.
 	// +optional
-	Containers *ClContainerSelector `json:"containers"`
+	Containers *ClContainerSelector `json:"containers,omitempty"`
 
 	// direction specifies the direction of traffic the tc program should
 	// attach to for a given network device.
+	// +optional
```

Comment:
Why did you make these optional?  I think direction is required for both tc and tcx.  Priority is required for tc and xdp.  (same question for all the program types.)

Reply:
I just revert the direction but pri when we have interface discovery we need to use default pri for all programs